indico/modules/events/editing/client/js/components/FileDisplay/index.jsx

```diff
@@ -0,0 +1,55 @@
+import React, {useMemo} from 'react';
+import PropTypes from 'prop-types';
+import {fileTypePropTypes, filePropTypes, mapFileTypes} from '../FileManager/util';
+import './FileDisplay.module.scss';
+
+function FileListDisplay({files}) {
+  return (
+    <ul styleName="file-list-display">
+      {files.map(({filename, uuid}) => (
+        <li key={uuid}>
+          <span styleName="file-name">
+            <a>{filename}</a>
+          </span>
+        </li>
+      ))}
+    </ul>
+  );
+}
+
+FileListDisplay.propTypes = {
+  files: PropTypes.arrayOf(PropTypes.shape(filePropTypes)).isRequired,
+};
+
+function FileTypeDisplay({fileType}) {
+  return (
+    <div styleName="file-type-display">
+      <h3>{fileType.name}</h3>
+      <FileListDisplay
+        files={fileType.files}
+        fileTypeId={fileType.id}
+        multiple={fileType.multiple}
+      />
+    </div>
+  );
+}
+
+FileTypeDisplay.propTypes = {
+  fileType: PropTypes.shape(fileTypePropTypes).isRequired,
+};
+
+export default function FileDisplay({fileTypes, files}) {
+  const _fileTypes = useMemo(() => mapFileTypes(fileTypes, files), [fileTypes, files]);
```

Comment:
I am wondering whether the benefit of `useMemo` is enough to justify its usage here. This component probably won't rerender so often, right? It doesn't hurt to use it but I am just thinking whether it is so critical here